x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentForInitialNode.java

```diff
@@ -0,0 +1,196 @@
+                                                 CheckedFunction<Environment, CreateEnrollmentToken, Exception>
+                                                     createEnrollmentTokenFunction){
+        super(clientFunction, keyStoreFunction, "Set elastic password and generate enrollment token for initial node");
+        this.createEnrollmentTokenFunction = createEnrollmentTokenFunction;
+        parser.allowsUnrecognizedOptions();
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new BootstrapPasswordAndEnrollmentForInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (options.nonOptionArguments().contains("--explicitly-acknowledge-execution") == false) {
+            throw new UserException(ExitCodes.CONFIG, "This command is not intended for end users");
+        }
+        if (env.settings().hasValue(XPackSettings.ENROLLMENT_ENABLED.getKey()) && false ==
+            XPackSettings.ENROLLMENT_ENABLED.get(env.settings())) {
+            throw new UserException(ExitCodes.NOOP, "Enrollment is explicitly disabled.");
+        }
+        final CommandLineHttpClient client = getClient(env);
+        final CreateEnrollmentToken createEnrollmentToken = getCreateEnrollmentToken(env);
+        keyStoreWrapper = getKeyStoreWrapper(env);
+        ReadBootstrapPassword(env, terminal);
+        checkClusterHealthWithRetries(client, terminal, elasticUser, credentialsPassword, 0, 5, false);
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            changeElasticUserPassword(terminal, client);
+        } else {
+            password = bootstrapPassword;
+        }
+        token = createEnrollmentToken.createKibanaEnrollmentToken(elasticUser, password);
+        fingerprint = createEnrollmentToken.getFingerprint();
+
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            terminal.println("'elastic' user password: " + password);
+        }
+        terminal.println("CA fingerprint: " + fingerprint);
+        terminal.println("Kibana enrollment token: " + token);
+        if (options.nonOptionArguments().contains("--docker")) {
+            nodeToken = createEnrollmentToken.createNodeEnrollmentToken(elasticUser, password);
+            terminal.println("Node enrollment token: " + nodeToken);
+        }
+    }
+
+    @Override
+    public void close() {
+        if (keyStoreWrapper != null) {
+            keyStoreWrapper.close();
+        }
+        if (bootstrapPassword != null) {
+            bootstrapPassword.close();
+        }
+        if (credentialsPassword != null) {
+            credentialsPassword.close();
+        }
+    }
+
+    protected void changeElasticUserPassword(Terminal terminal, CommandLineHttpClient client) throws Exception {
```

Comment:
can we have this return the password instead of being void and setting a value to a class var ?